test/gold/notes.clang.-g-grecord-gcc-switches-O1-O1-fno-discard-value-names.1-callsite.subset.operand_points_to.golden.csv

```diff
@@ -85,18 +85,18 @@ nil	*global_alloc@.str.4[0]	[<notes.c>:handle:25, nil]	<notes.c>:notes_strdup:%s
 nil	*global_alloc@.str.4[0]	nil	<notes.c>:handle:25:1:getelementptr inbounds ([17 x i8], [17 x i8]* @.str.4, i64 0, i64 0)
 nil	*global_alloc@.str.4[0][*]	[<notes.c>:handle:25, nil]	<notes.c>:notes_strdup:%s
 nil	*global_alloc@.str.4[0][*]	nil	<notes.c>:handle:25:1:getelementptr inbounds ([17 x i8], [17 x i8]* @.str.4, i64 0, i64 0)
-nil	*global_alloc@.str.5	nil	<notes.c>:handle_loop:12:2:@.str.5
 nil	*global_alloc@.str.5	nil	<notes.c>:handle_loop:2:2:@.str.5
-nil	*global_alloc@.str.5[0]	nil	<notes.c>:handle_loop:12:1:getelementptr inbounds ([8 x i8], [8 x i8]* @.str.5, i64 0, i64 0)
+nil	*global_alloc@.str.5	nil	<notes.c>:handle_loop:36:2:@.str.5
 nil	*global_alloc@.str.5[0]	nil	<notes.c>:handle_loop:2:1:getelementptr inbounds ([8 x i8], [8 x i8]* @.str.5, i64 0, i64 0)
-nil	*global_alloc@.str.5[0][*]	nil	<notes.c>:handle_loop:12:1:getelementptr inbounds ([8 x i8], [8 x i8]* @.str.5, i64 0, i64 0)
+nil	*global_alloc@.str.5[0]	nil	<notes.c>:handle_loop:36:1:getelementptr inbounds ([8 x i8], [8 x i8]* @.str.5, i64 0, i64 0)
 nil	*global_alloc@.str.5[0][*]	nil	<notes.c>:handle_loop:2:1:getelementptr inbounds ([8 x i8], [8 x i8]* @.str.5, i64 0, i64 0)
-nil	*global_alloc@.str.6	nil	<notes.c>:handle_loop:21:2:@.str.6
-nil	*global_alloc@.str.6[0]	nil	<notes.c>:handle_loop:21:1:getelementptr inbounds ([16 x i8], [16 x i8]* @.str.6, i64 0, i64 0)
-nil	*global_alloc@.str.6[0][*]	nil	<notes.c>:handle_loop:21:1:getelementptr inbounds ([16 x i8], [16 x i8]* @.str.6, i64 0, i64 0)
-nil	*global_alloc@.str.7	nil	<notes.c>:handle_loop:43:2:@.str.7
-nil	*global_alloc@.str.7[0]	nil	<notes.c>:handle_loop:43:1:getelementptr inbounds ([2 x i8], [2 x i8]* @.str.7, i64 0, i64 0)
-nil	*global_alloc@.str.7[0][*]	nil	<notes.c>:handle_loop:43:1:getelementptr inbounds ([2 x i8], [2 x i8]* @.str.7, i64 0, i64 0)
+nil	*global_alloc@.str.5[0][*]	nil	<notes.c>:handle_loop:36:1:getelementptr inbounds ([8 x i8], [8 x i8]* @.str.5, i64 0, i64 0)
+nil	*global_alloc@.str.6	nil	<notes.c>:handle_loop:10:2:@.str.6
+nil	*global_alloc@.str.6[0]	nil	<notes.c>:handle_loop:10:1:getelementptr inbounds ([16 x i8], [16 x i8]* @.str.6, i64 0, i64 0)
+nil	*global_alloc@.str.6[0][*]	nil	<notes.c>:handle_loop:10:1:getelementptr inbounds ([16 x i8], [16 x i8]* @.str.6, i64 0, i64 0)
+nil	*global_alloc@.str.7	nil	<notes.c>:handle_loop:32:2:@.str.7
+nil	*global_alloc@.str.7[0]	nil	<notes.c>:handle_loop:32:1:getelementptr inbounds ([2 x i8], [2 x i8]* @.str.7, i64 0, i64 0)
+nil	*global_alloc@.str.7[0][*]	nil	<notes.c>:handle_loop:32:1:getelementptr inbounds ([2 x i8], [2 x i8]* @.str.7, i64 0, i64 0)
```

Comment:
Triage: There's one more addition than deletion here, otherwise everything seems related to instruction ordering changes or function renamings.

Reply:
<details>
<summary>Comparison script</summary>

```python
from pathlib import Path
from subprocess import PIPE, check_output, run
from sys import stdin, stdout, stderr

BASH = """
sudo rm -rf 10 11
docker_here() {
    docker run --rm -it --mount type=bind,src=$PWD,target=/work --workdir /work "${@}"
}
docker_here cclyzer-dev-llvm-11 clang-11 -O1 -O1 -fno-discard-value-names -S -emit-llvm test/c/cfg-test.c -o 11.ll
docker pull ghcr.io/galoisinc/cclyzerpp-dev:main
docker tag ghcr.io/galoisinc/cclyzerpp-dev:main cclyzer-dev
docker_here cclyzer-dev clang-10 -O1 -O1 -fno-discard-value-names -S -emit-llvm test/c/cfg-test.c -o 10.ll
docker_here cclyzer-dev-llvm-11 cmake -G Ninja -S . -B build -DLLVM_VERSION=11 -DCLANG_VERSION=11
docker_here cclyzer-dev-llvm-11 cmake --build build -j $(nproc) --target factgen-exe
docker_here cclyzer-dev-llvm-11 ./build/factgen-exe 11.ll --out-dir 11
sudo chown -R $(whoami):$(whoami) 11
for f in 11/*.gz; do gunzip "${f}"; done
for f in 11/*.csv; do sort "${f}" > "${f}.sorted"; rm "${f}"; done
docker_here cclyzer-dev cmake -G Ninja -S . -B build-10 -DLLVM_VERSION=10 -DCLANG_VERSION=10
docker_here cclyzer-dev cmake --build build-10 -j $(nproc) --target factgen-exe
docker_here cclyzer-dev ./build/factgen-exe 10.ll --out-dir 10
sudo chown -R $(whoami):$(whoami) 10
for f in 10/*.gz; do gunzip "${f}"; done
for f in 10/*.csv; do sort "${f}" > "${f}.sorted"; rm "${f}"; done
pushd 10
sed_recurse() { ag -g '.*' -0 | xargs -0 sed -E -i "$@"; }
sed_recurse 's|10\.ll|11\.ll|g'
popd
for f in 10/*.csv.sorted; do diff -u "${f}" 11/$(basename ${f}); done
"""

TEMPLATE = """
<details>
<summary>Comparison script</summary>

```python
{script}
```

</details>
<details>
<summary>10.ll</summary>

```llvm
{ten}
```

</details>

<details>
<summary>11.ll</summary>

```llvm
{eleven}
```

</details>

<details>
<summary>LLVM diff</summary>

```diff
{lldiff}
```

</details>

<details>
<summary>Relation diff</summary>

```diff
{lldiff}
```

</details>
"""

if __name__ == "__main__":
    Path("script.sh").write_text(BASH)
    out = check_output(["bash", "script.sh"], stdin=stdin, stderr=stderr)
    print(TEMPLATE.format(
        script=Path(__file__).read_text(),
        ten=Path("10.ll").read_text(),
        eleven=Path("11.ll").read_text(),
        lldiff=run(["diff", "-u", "10.ll", "11.ll"], stdout=PIPE).stdout.decode('utf-8'),
        reldiff=out,
    ))

```

</details>
<details>
<summary>10.ll</summary>

```llvm
; ModuleID = 'test/c/cfg-test.c'
source_filename = "test/c/cfg-test.c"
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-linux-gnu"

@global = dso_local local_unnamed_addr global i32 0, align 4

; Function Attrs: norecurse nounwind readnone uwtable
define dso_local i32 @bar(i32 returned %x) local_unnamed_addr #0 {
entry:
  ret i32 %x
}

; Function Attrs: norecurse nounwind readonly uwtable
define dso_local i32 @foo(i32 %x) local_unnamed_addr #1 {
entry:
  %call = call i32 @bar(i32 %x)
  %add = shl i32 %call, 1
  %0 = load i32, i32* @global, align 4, !tbaa !2
  %add2 = add nsw i32 %add, %0
  ret i32 %add2
}

; Function Attrs: nofree norecurse nounwind uwtable writeonly
define dso_local void @baz() local_unnamed_addr #2 {
entry:
  store i32 1, i32* @global, align 4, !tbaa !2
  ret void
}

; Function Attrs: nofree norecurse nounwind uwtable
define dso_local i32 @main(i32 %argc, i8** nocapture readnone %argv) local_unnamed_addr #3 {
entry:
  %call = call i32 @foo(i32 0)
  call void @baz()
  %call1 = call i32 @foo(i32 1)
  %call2 = call i32 @bar(i32 %call1)
  %add = add nsw i32 %call1, %call
  %add3 = add nsw i32 %add, %call2
  ret i32 %add3
}

attributes #0 = { norecurse nounwind readnone uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
attributes #1 = { norecurse nounwind readonly uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
attributes #2 = { nofree norecurse nounwind uwtable writeonly "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
attributes #3 = { nofree norecurse nounwind uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }

!llvm.module.flags = !{!0}
!llvm.ident = !{!1}

!0 = !{i32 1, !"wchar_size", i32 4}
!1 = !{!"clang version 10.0.0-4ubuntu1 "}
!2 = !{!3, !3, i64 0}
!3 = !{!"int", !4, i64 0}
!4 = !{!"omnipotent char", !5, i64 0}
!5 = !{!"Simple C/C++ TBAA"}

```

</details>

<details>
<summary>11.ll</summary>

```llvm
; ModuleID = 'test/c/cfg-test.c'
source_filename = "test/c/cfg-test.c"
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-linux-gnu"

@global = dso_local local_unnamed_addr global i32 0, align 4

; Function Attrs: norecurse nounwind readnone uwtable
define dso_local i32 @bar(i32 returned %x) local_unnamed_addr #0 {
entry:
  ret i32 %x
}

; Function Attrs: norecurse nounwind readonly uwtable
define dso_local i32 @foo(i32 %x) local_unnamed_addr #1 {
entry:
  %add = shl nsw i32 %x, 1
  %0 = load i32, i32* @global, align 4, !tbaa !2
  %add2 = add nsw i32 %0, %add
  ret i32 %add2
}

; Function Attrs: nofree norecurse nounwind uwtable writeonly
define dso_local void @baz() local_unnamed_addr #2 {
entry:
  store i32 1, i32* @global, align 4, !tbaa !2
  ret void
}

; Function Attrs: nofree norecurse nounwind uwtable
define dso_local i32 @main(i32 %argc, i8** nocapture readnone %argv) local_unnamed_addr #3 {
entry:
  %call = call i32 @foo(i32 0)
  call void @baz()
  %call1 = call i32 @foo(i32 1)
  %reass.add = shl i32 %call1, 1
  %add3 = add i32 %reass.add, %call
  ret i32 %add3
}

attributes #0 = { norecurse nounwind readnone uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
attributes #1 = { norecurse nounwind readonly uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
attributes #2 = { nofree norecurse nounwind uwtable writeonly "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
attributes #3 = { nofree norecurse nounwind uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }

!llvm.module.flags = !{!0}
!llvm.ident = !{!1}

!0 = !{i32 1, !"wchar_size", i32 4}
!1 = !{!"Ubuntu clang version 11.0.0-2~ubuntu20.04.1"}
!2 = !{!3, !3, i64 0}
!3 = !{!"int", !4, i64 0}
!4 = !{!"omnipotent char", !5, i64 0}
!5 = !{!"Simple C/C++ TBAA"}

```

</details>

<details>
<summary>LLVM diff</summary>

```diff
--- 10.ll	2022-10-25 15:35:40.421378825 -0400
+++ 11.ll	2022-10-25 15:35:39.632370481 -0400
@@ -14,10 +14,9 @@
 ; Function Attrs: norecurse nounwind readonly uwtable
 define dso_local i32 @foo(i32 %x) local_unnamed_addr #1 {
 entry:
-  %call = call i32 @bar(i32 %x)
-  %add = shl i32 %call, 1
+  %add = shl nsw i32 %x, 1
   %0 = load i32, i32* @global, align 4, !tbaa !2
-  %add2 = add nsw i32 %add, %0
+  %add2 = add nsw i32 %0, %add
   ret i32 %add2
 }
 
@@ -34,22 +33,21 @@
   %call = call i32 @foo(i32 0)
   call void @baz()
   %call1 = call i32 @foo(i32 1)
-  %call2 = call i32 @bar(i32 %call1)
-  %add = add nsw i32 %call1, %call
-  %add3 = add nsw i32 %add, %call2
+  %reass.add = shl i32 %call1, 1
+  %add3 = add i32 %reass.add, %call
   ret i32 %add3
 }
 
-attributes #0 = { norecurse nounwind readnone uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
-attributes #1 = { norecurse nounwind readonly uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
-attributes #2 = { nofree norecurse nounwind uwtable writeonly "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
-attributes #3 = { nofree norecurse nounwind uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
+attributes #0 = { norecurse nounwind readnone uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
+attributes #1 = { norecurse nounwind readonly uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
+attributes #2 = { nofree norecurse nounwind uwtable writeonly "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
+attributes #3 = { nofree norecurse nounwind uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
 
 !llvm.module.flags = !{!0}
 !llvm.ident = !{!1}
 
 !0 = !{i32 1, !"wchar_size", i32 4}
-!1 = !{!"clang version 10.0.0-4ubuntu1 "}
+!1 = !{!"Ubuntu clang version 11.0.0-2~ubuntu20.04.1"}
 !2 = !{!3, !3, i64 0}
 !3 = !{!"int", !4, i64 0}
 !4 = !{!"omnipotent char", !5, i64 0}

```

</details>

<details>
<summary>Relation diff</summary>

```diff
--- 10.ll	2022-10-25 15:35:40.421378825 -0400
+++ 11.ll	2022-10-25 15:35:39.632370481 -0400
@@ -14,10 +14,9 @@
 ; Function Attrs: norecurse nounwind readonly uwtable
 define dso_local i32 @foo(i32 %x) local_unnamed_addr #1 {
 entry:
-  %call = call i32 @bar(i32 %x)
-  %add = shl i32 %call, 1
+  %add = shl nsw i32 %x, 1
   %0 = load i32, i32* @global, align 4, !tbaa !2
-  %add2 = add nsw i32 %add, %0
+  %add2 = add nsw i32 %0, %add
   ret i32 %add2
 }
 
@@ -34,22 +33,21 @@
   %call = call i32 @foo(i32 0)
   call void @baz()
   %call1 = call i32 @foo(i32 1)
-  %call2 = call i32 @bar(i32 %call1)
-  %add = add nsw i32 %call1, %call
-  %add3 = add nsw i32 %add, %call2
+  %reass.add = shl i32 %call1, 1
+  %add3 = add i32 %reass.add, %call
   ret i32 %add3
 }
 
-attributes #0 = { norecurse nounwind readnone uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
-attributes #1 = { norecurse nounwind readonly uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
-attributes #2 = { nofree norecurse nounwind uwtable writeonly "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
-attributes #3 = { nofree norecurse nounwind uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
+attributes #0 = { norecurse nounwind readnone uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
+attributes #1 = { norecurse nounwind readonly uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
+attributes #2 = { nofree norecurse nounwind uwtable writeonly "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
+attributes #3 = { nofree norecurse nounwind uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="none" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
 
 !llvm.module.flags = !{!0}
 !llvm.ident = !{!1}
 
 !0 = !{i32 1, !"wchar_size", i32 4}
-!1 = !{!"clang version 10.0.0-4ubuntu1 "}
+!1 = !{!"Ubuntu clang version 11.0.0-2~ubuntu20.04.1"}
 !2 = !{!3, !3, i64 0}
 !3 = !{!"int", !4, i64 0}
 !4 = !{!"omnipotent char", !5, i64 0}

```

</details>

Reply:
The structure of the code changes considerably from LLVM 10 to LLVM 11. I'm calling this OK.